### PR TITLE
fix autocompletion on OSX

### PR DESCRIPTION
**Pull request by theY4Kman**
Here's the patch to fix autocompletion on OSX, ref antocuni/pdb#74

### CODE REVIEW

**Comment by blueyed** [comment]
Thank you!

**Comment by blueyed** [comment]
btw: I plan to have an integration test for this (in the pdbpp repo probably), using MacOS on Travis.
Can you provide more details maybe about what is required to trigger this, or should it be the case always with MacOS?

File: fancycompleter.py [hunk 1]
```diff
@@ -336,19 +336,15 @@ def commonfunc(s1,s2):
...
     if config.using_pyrepl or sys.platform != 'darwin':
         return False
-    cmd =  "otool -L %s | grep libedit" % config.readline.__file__
-    (status, result) = commands.getstatusoutput(cmd)
-    if status == 0 and len(result) > 0:
+    from subprocess import Popen, PIPE
+    cmd = ["otool", "-L", config.readline.__file__]
+    p = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = p.communicate()
+    if p.returncode == 0 and 'libedit' in stdout.decode('utf-8'):
```
Comment (blueyed) [inline fancycompleter.py]: btw: I've wondered if there is something in `os.environ` that could be looked at instead / or as a first (faster) check.

**Comment by theY4Kman** [comment]
I'm not entirely sure what triggers it. It does seem to happen every time on my MBP w/ Mojave (10.14.2). I'll play around and see if I can drum up more useful information.

**Comment by theY4Kman** [comment]
Ooo, one thing to note: IPython's [`rlineimpl.py`](https://github.com/ipython/rlipython/blob/e218435e396a5798e001f732afa5b5e5fdc93ce0/rlipython/rlineimpl.py) doesn't use (or no longer uses) the otool method. Instead, they check for existence of "libedit" in the `readline` module's docstring, because
> Official Python docs state that 'libedit' is in the docstring for libedit readline

Their implementation is basically
```py
for readline_mod_name in ['readline', 'gnureadline']:
    try:
        readline = __import__(readline_mod_name)
        uses_libedit = readline.__doc__ and 'libedit' in readline.__doc__

        if sys.platform == "darwin" and uses_libedit:
            # System / Python.org pythons should use gnureadline
            print("will try to use gnureadline")
            continue
    except ImportError:
        pass
```

Changing `has_leopard_libedit` to this implementation seems to work as well as the patch in this PR, from cursory red-green manual tests on me machine:
```py
def has_leopard_libedit(config):
    # Detect if we are using Leopard's libedit.
    # Adapted from IPython's rlineimpl.py.
    if config.using_pyrepl or sys.platform != 'darwin':
        return False
    return config.readline.__doc__ and 'libedit' in config.readline.__doc__
```

**Comment by blueyed** [comment]
Awesome, please create a PR for/with it.

**Pull request merged**